internal/renderer/renderer_test.go

```diff
@@ -0,0 +1,95 @@
+package renderer
+
+import (
+	"testing"
+
+	"github.com/docker/app/internal/renderer/driver"
+	"github.com/pkg/errors"
+	"gotest.tools/assert"
+	is "gotest.tools/assert/cmp"
+)
+
+type fakeDriver struct{}
+
+func (d *fakeDriver) Apply(s string, settings map[string]interface{}) (string, error) {
+	return s + "fake", nil
+}
+
+type fakeErrorDriver struct{}
+
+func (d *fakeErrorDriver) Apply(s string, settings map[string]interface{}) (string, error) {
+	return s, errors.New("error in driver")
+}
+
+func TestRegisterNilPanics(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+		resetDrivers()
+	}()
+	Register("foo", nil)
+}
+
+func TestRegisterDuplicatePanics(t *testing.T) {
+	defer func() {
+		if r := recover(); r == nil {
+			t.Errorf("The code did not panic")
+		}
+		resetDrivers()
+	}()
+	Register("bar", &fakeDriver{})
+	Register("bar", &fakeDriver{})
+}
+
+func TestRegister(t *testing.T) {
+	fd := &fakeDriver{}
```

Comment:
nit: `fd` has a specific meaning, here I believe `d` might be more appropriate

Reply:
You mean file descriptor? I think in this context one does not presume there is any file descriptor here, am I wrong?